### PR TITLE
Modify rule S6535: Adapt to LaYC

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**

### CODE REVIEW

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
Let's move most of introduction to the first section

File: rules/S6535/javascript/rule.adoc [hunk 1]
```diff
@@ -1,24 +1,24 @@
-In JavaScript, we can use the ``++/++`` (backslash) character to escape special characters inside a string, a template literal, or a regular expression.
+== Why is this an issue?
+
+In JavaScript, the ``++/++`` (backslash) character is used to escape special characters inside a string, a template literal, or a regular expression.
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:3]: I am not sure why we need a new line for each sentence. It does not look very pretty. I would merge all of them into one paragraph

File: rules/S6535/javascript/rule.adoc [hunk 2]
```diff
@@ -28,7 +28,7 @@ const path  = `\/${some}\/${dir}`;
 
 ==== Compliant solution
 
-[source,javascript]
+[source,javascript,diff-id=1,diff-type=compliant]
 ----
 const regex = /[[]/;
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:33]: I think some link to the docs about which characters are special and could be escaped would not hurt

File: rules/S6535/javascript/rule.adoc [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+Useless character escapes can safely be removed without changing the original behavior.
 
 === Code examples
 
 ==== Noncompliant code example
 
-[source,javascript]
+[source,javascript,diff-id=1,diff-type=noncompliant]
 ----
 const regex = /[\[]/;
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:23]: could you add a comment here explaining why it's not compliant? `[` seems to be a special char in regex

File: rules/S6535/javascript/rule.adoc [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 const octal = '\8';
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:24]: and here :D

File: rules/S6535/javascript/rule.adoc [hunk 5]
```diff
@@ -1,24 +1,24 @@
-In JavaScript, we can use the ``++/++`` (backslash) character to escape special characters inside a string, a template literal, or a regular expression.
+== Why is this an issue?
+
+In JavaScript, the ``++/++`` (backslash) character is used to escape special characters inside a string, a template literal, or a regular expression.
 
 It indicates that the next character should be treated as a literal character rather than as a special character or string delimiter.
 
 For instance, it is common to escape single quotes inside a string literal using the single quote delimiter like ``++'It\'s a beautiful day'++``.
 
 Escaping is only meaningful for special characters.
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:9]: I think this phrase can be dropped, it does not bring much

File: rules/S6535/javascript/rule.adoc [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
 
-== Why is this an issue?
-
 Escaping non-special characters in strings, template literals, and regular expressions doesn't affect their value. Therefore, useless escapes impact code readability and could even denote a bug in the code if the developer left it by mistake or intended to escape another special character instead.
 
 == How to fix it
 
-We can safely remove a useless character escape without changing the original behaviour.
+Useless character escapes can safely be removed without changing the original behavior.
```
Comment (vilchik-elena) [inline rules/S6535/javascript/rule.adoc:15]: how about adding a mention that the developer should first make sure that the intention was to simply use the escaped char and that there is no hidden bug.

**Review by vilchik-elena — CHANGES_REQUESTED** [review]

File: rules/S6535/javascript/rule.adoc (hunk 2 above)
Comment (ilia-kebets-sonarsource) [reply rules/S6535/javascript/rule.adoc:28]: There is no exhaustive link, I have added 2.

**Pull request closed without merging**